backend/src/scripts/utility/users/identifyOrphanedUsers.js

```diff
@@ -0,0 +1,98 @@
+/**
+ *  Copyright (C) 2025 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const Path = require('path');
+const { v5Path } = require('../../../interop');
+const FS = require('fs');
+
+const { logger } = require(`${v5Path}/utils/logger`);
+const { getUsersByQuery } = require(`${v5Path}/models/users`);
+const { getLastLoginDate } = require(`${v5Path}/models/loginRecords`);
+
+const DEFAULT_OUT_FILE = 'orphanedUsers.csv';
+
+const writeResultsToFile = (results, outFile) => new Promise((resolve) => {
+	logger.logInfo(`Writing results to ${outFile}`);
+	const writeStream = FS.createWriteStream(outFile);
+	writeStream.write('Username,First Name,Last Name,Email,Company,Last Login\n');
+	results.forEach(({ user, firstName, lastName, email, company, lastLogin }) => {
+		writeStream.write(`${user},${firstName},${lastName},${email},${company},${lastLogin}\n`);
+	});
+
+	writeStream.end(resolve);
+});
+
+const getFileEntry = async ({ user, customData }) => {
+	const lastLogin = await getLastLoginDate(user);
+	const { firstName, lastName, email, billing: { billingInfo: { company } } } = customData;
+
+	return { user, firstName, lastName, email, company: company ?? '', lastLogin: lastLogin ?? '' };
+};
+
+const run = async (outFile) => {
+	const query = {
```

Comment:
This seems a bit overkill, can't we just query the array?

this seems to work? 
![image](https://github.com/user-attachments/assets/e84cdc17-483c-4c39-9f0f-6ba591b41cd0)